app/dmps/[...slug]/page.tsx

```diff
@@ -0,0 +1,330 @@
+import { gql, useQuery } from '@apollo/client';
+import createApolloClient from "@/apollo-client";
+import {headers} from 'next/headers';
+import {redirect} from 'next/navigation';
+import Image from 'next/image';
+import narrativeLogo from '@/public/images/u153.svg';
+
+import { DmptoolLink, Link } from '@/components/link/link';
+
+import Abstract from "@/components/abstract";
+import Citation from "@/components/citation";
+import Contributors from "@/components/contributors";
+import Footer from "@/components/footer";
+import Funding from "@/components/funding";
+import Outputs from "@/components/outputs";
+import Project from "@/components//project";
+import Versions from "@/components/versions";
+import Works from "@/components//works";
+import '../[...slug]/dmps.scss';
+
+const client = createApolloClient();
+
+interface RelatedIdentifier {
+    descriptor: string;
+    work_type: string;
+    identifier: string
+}
+
+interface ContactInterface {
+    name: string
+}
+interface FormData {
+    json_url: string,
+    title: string,
+    description: string,
+    dmp_id: string,
+    privacy: string,
+    created: string,
+    modified: string,
+    ethical_issues_exist: string,
+    ethical_issues_report: string,
+
+    funder_name: string,
+    funder_id: string,
+    award_id: string,
+    opportunity_number: string,
+
+    project_title: string,
+    project_abstract: string,
+    project_start: string,
+    project_end: string,
+
+    contact: ContactInterface | {},
+    contributors: String[],
+    datasets: String[],
+    related_identifiers: RelatedIdentifier[],
+    versions: String[],
+}
+
+function getUrl() {
+    const headersList = headers();
+    const header_url = headersList.get('x-url') || "";
+    const url = new URL(header_url);
+    let apiHost = (url.hostname === 'localhost') ? 'api.dmphub.uc3dev.cdlib.net' : `api.${url.hostname}`;
```

Comment:
Isn't this duplicating the code in api.ts?

Reply:
Yea, it's a bit confusing. I was trying to see if I could generate the page on the server side, and the server side does not have access to the "window" browser object, so I needed to grab the same info about the host from the header.

Reply:
ah ok. that makes sense. I had forgotten about the server side/client side division